dev/devicelab/lib/tasks/microbenchmarks.dart

```diff
@@ -95,5 +97,11 @@ Future<Map<String, double>> _readJsonResults(Process process) {
       jsonBuf.writeln(line.substring(prefixLength));
   });
 
+  process.exitCode.then((int code) {
```

Comment:
This is intended for cases where "flutter run" dies due to an error.  Ordinarily it will run until it produces the output JSON, at which point the microbenchmark will kill it.

Changed this to complete the future with an error (instead of throwing and relying on a toplevel uncaught exception handler)